klippy/heater.py

```diff
@@ -228,12 +228,14 @@ def __init__(self, config):
                                desc=self.cmd_TURN_OFF_HEATERS_help)
     def add_sensor(self, sensor_type, sensor_factory):
         self.sensors[sensor_type] = sensor_factory
-    def setup_heater(self, config, gcode_id):
-        heater_name = config.get_name()
+    def setup_heater(self, config, gcode_id, heater_name=None):
+        if heater_name is None:
+            heater_name = config.get_name()
```

Comment:
I think if we changed the existing code to `heater_name = config.get_name().split()[-1]` and did the same in the Heater() class, then we would not need any special handling of names in other parts of the code.